### DISCUSSION TITLE
error: failed to update dependencies

### DISCUSSION THREAD

**Discussion by luisvivasg** [deta/cloud-docs#631] [Help]
I am trying to update the depandancies in the requirement.txt. It has been working fine but sudeenly stopped working. I saw in the website, it says to make sure it is in UTF-8 format, I have checked and recreated the file and I still get the same error. I have also tried to delete packages and get the same result using python.  Any suggestion?

My files only contains this:
[requirements.txt](https://github.com/deta/docs/files/10171166/requirements.txt)

**Comment by lemonyte** [comment]
1. `pandas` may be too large to fit in the 250mb size limit (check the size of your venv)
2. `subprocess.run` is a legacy extension for Python 2 and Python 3.3, you probably don't need it
3. Do you need both `flask` and `fastapi` at the same time?

**Reply by luisvivasg** [reply]
Hello Lemon. Thanks for your answer.

1. I have 512mb Padas was installed correclty at the beggining.
2. I just using it since I want to run some things for testing in the server. Unless, there is way to access the terminal, in the Deta documentation I could not find anything.
3. True I don't need it. I was testing and forgot to remove it.

I deteled the micro and created a new one. It seems there is a bug. I was adding packages one by one and it was working fine. I added pandas jsut to see the capacity and owrked fine. Then I added subprocess.run, I said: "Error: failed to update dependencies: Internal server error". I tried again in was installed. Then added Pickle just to test. And boom: "Error: failed to remove dependecies: error on one or more dependencies, no dependencies were removed, see output for details"

It looks it is what you mentioned. I am collapsing the size limit but the error does not correspond to the message and it is not related to the solution given regaring the UTF-8. The problem now is that I cannot revert. When I try to remove the packages even if I only leave Deta, it is going to show me the same message. The only solution is to remove and create a new micro. This needs a fix. It would be very good to have a way to know the usage so it is easier to know when you are reaching the limit. 

Regards.

**Reply by lemonyte** [reply]
I agree that the error messages are not descriptive, that is something the team is working on improving. Like I mentioned, `subprocess.run` is a legacy extension for Python 2 and 3.3, newer versions have a `run` function [built-in](https://docs.python.org/3/library/subprocess.html#subprocess.run). It seems that pip is failing to uninstall `subprocess.run` because of some compatibility issue. Try recreating the Micro from scratch, but without `subprocess.run` in the dependencies.
